openmeter/app/service.go

```diff
@@ -0,0 +1,30 @@
+package app
+
+import (
+	"context"
+
+	appentity "github.com/openmeterio/openmeter/openmeter/app/entity"
+	appentitybase "github.com/openmeterio/openmeter/openmeter/app/entity/base"
+	"github.com/openmeterio/openmeter/pkg/pagination"
+)
+
+type Service interface {
+	AppService
+}
+
+type AppService interface {
+	// Marketplace
+	RegisterMarketplaceListing(input appentity.RegisterMarketplaceListingInput) error
+	GetMarketplaceListing(ctx context.Context, input appentity.MarketplaceGetInput) (appentity.RegistryItem, error)
+	ListMarketplaceListings(ctx context.Context, input appentity.MarketplaceListInput) (pagination.PagedResponse[appentity.RegistryItem], error)
+	InstallMarketplaceListingWithAPIKey(ctx context.Context, input appentity.InstallAppWithAPIKeyInput) (appentity.App, error)
+	GetMarketplaceListingOauth2InstallURL(ctx context.Context, input appentity.GetOauth2InstallURLInput) (appentity.GetOauth2InstallURLOutput, error)
+	AuthorizeMarketplaceListingOauth2Install(ctx context.Context, input appentity.AuthorizeOauth2InstallInput) error
+
+	// Installed app
+	CreateApp(ctx context.Context, input appentity.CreateAppInput) (appentitybase.AppBase, error)
+	GetApp(ctx context.Context, input appentity.GetAppInput) (appentity.App, error)
+	GetDefaultApp(ctx context.Context, input appentity.GetDefaultAppInput) (appentity.App, error)
```

Comment:
This is more theoretical, but it is a decision how we want to do the defaulting.

Right now we have a default app per type, which allows us to support multiple apps from the same type, while not mandating the partner to store the application id.

I wonder if it makes sense to have a default app per capability (e.g. tax reporting, etc).

If not we should be prudent documenting the default behavior.

Reply:
It's perfectly logical and fine to have multiple "apps" of the same type. AWS/GCP comes to mind for object storage export.

Reply:
We discussed that we keep per app type default to make onboarding a smoother experience.